app/src/main/java/com/ataulm/artcollector/gallery/ui/UiGallery.kt

```diff
@@ -0,0 +1,3 @@
+package com.ataulm.artcollector.gallery.ui
+
+internal class UiGallery(collection: Collection<UiPainting>) : ArrayList<UiPainting>(collection)
```

Comment:
I was considering ditching this class (and just use `List<UiPainting>`), but I kept it.

In a subsequent PR (or attempt at one), I want to introduce the Android Paging library (PagedLists). (These last two PRs have been intentionally crafted to make it more difficult to introduce Android Paging Library.)

- I want to maintain the distinction between domain and UI models - the examples I've seen so far in Paging Library demos use one model across layers of the app
- ideally I don't want Android classes, like PagedList or LiveData, in my domain because keeping domain modules pure-Kotlin leaves the door open for sharing code between platforms (#pipedream)
- I want minimal coupling between the View/UI layer and the domain/data layers though I accept there may be some inherent with the concept of paging (e.g. page size can be informed by how many items we want to show at once/placeholders generated by the PagedList require support for nullable items in the adapter, etc.)
- I'm not using Room here yet, and even if I was, I'd want to treat that as an implementation detail; access to Room from my domain should go via a repository class/interface that doesn't have a dependency on android libraries.